components/RefreshRemoteUser.coffee

```diff
@@ -0,0 +1,66 @@
+noflo = require 'noflo'
+
+exports.getComponent = ->
+  c = new noflo.Component
+  c.inPorts.add 'in',
+    datatype: 'object'
+  c.outPorts.add 'pass',
+    datatype: 'object'
+  c.outPorts.add 'updated',
+    datatype: 'object'
+  c.outPorts.add 'invalid',
+    datatype: 'object'
+  c.outPorts.add 'error',
+    datatype: 'object'
+
+  noflo.helpers.WirePattern c,
+    in: 'in'
+    out: ['pass', 'updated', 'invalid']
+    async: true
+  , (data, groups, out, callback) ->
+    unless data?['grid-token']
+      # If user is not logged in, there is nothing to do
+      out.pass.send data
+      do callback
+      return
+
+    unless navigator.onLine
+      # When offline we can't refresh
+      out.pass.send data
```

Comment:
I'd consider sending the not-logged-in and not-online cases to different ports. That way the semantics of the component is visible from outside, in the graph.